src/pages/HabitTracker/Habits.tsx

```diff
@@ -0,0 +1,224 @@
+import "bootstrap/dist/css/bootstrap.min.css";
+import { useState, useEffect } from "react";
+import { Modal, Button, Form } from "react-bootstrap";
+import styled from "styled-components";
+import { PageBackground } from "../../components/styled/PageBackground";
+import { Sidebar } from "../../components/Sidebar";
+import { links } from "../../common/SidebarLinks";
+import { HabitTile } from "../../components/habittracker/HabitTile";
+import { AddHabitTile } from "../../components/habittracker/AddHabitTile";
+import { StyledModal } from "../../components/styled/Modal";
+
+// Styled component for the habit list container
+const HabitListContainer = styled.div`
+  width: calc(100% - 190px);
+  height: calc(100vh - 76px);
+  overflow-y: auto;
+  overflow-x: hidden;
+  position: absolute;
+  top: 56px;
+  left: 190px;
+
+  @media (max-width: 768px) {
+    width: calc(100% - 70px);
+    left: 70px;
+  }
+`;
+
+const HabitList = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  gap: 10px;
+  padding: 20px;
+  margin-left: 10px;
+  width: 100%;
+  box-sizing: border-box;
+  overflow-x: hidden;
+`;
+
+// Types
+interface Habit {
+  name: string;
+  count: number;
+  index: number;
+}
+
+// Functional component for Habits
+export function Habits() {
+  const [habits, setHabits] = useState<Habit[]>([]);
+  const [showModal, setShowModal] = useState(false);
+  const [newHabitName, setNewHabitName] = useState("");
+  const [newHabitCount, setNewHabitCount] = useState("");
+  const [editIndex, setEditIndex] = useState<number | null>(null);
+  const [originalHabitName, setOriginalHabitName] = useState("");
+  const [originalHabitCount, setOriginalHabitCount] = useState("");
```

Comment:
I would recommend to combine the states like:

```
type EditHabit = {
name: string;
count: number;
}

const [newHabit, setNewHabit] = useState<EditHabit>()
const [originalHabit, setOriginalHabit] = useState<EditHabit>()
```

Also, add small comments next to the states to clarify what the state is being used for.